cmake/kokkos_enable_options.cmake

```diff
@@ -58,6 +58,7 @@ KOKKOS_ENABLE_OPTION(IMPL_CUDA_UNIFIED_MEMORY OFF "Whether to leverage unified m
 KOKKOS_ENABLE_OPTION(DEPRECATED_CODE_4    ON "Whether code deprecated in major release 4 is available" )
 KOKKOS_ENABLE_OPTION(DEPRECATION_WARNINGS ON "Whether to emit deprecation warnings" )
 KOKKOS_ENABLE_OPTION(HIP_RELOCATABLE_DEVICE_CODE  OFF "Whether to enable relocatable device code (RDC) for HIP")
+KOKKOS_ENABLE_OPTION(IMPL_SYCL_GRAPH OFF "Whether to enable graph for SYCL (sycl_ext_oneapi_graph)")
```

Comment:
I don't think this is a good idea. We want support to be enabled if the implementation supports SYCL Command Graphs.

Reply:
I need a way to skip the `TestSYCL_InterOp_Graph.cpp` if there is no support for it.

But OK, I could just keep the `KOKKOS_IMPL_HAVE_SYCL_EXT_ONEAPI_GRAPH` and drop `KOKKOS_ENABLE_IMPL_SYCL_GRAPH` if you don't the new option. I don't feel strongly about it anyway.

Reply:
Still I think as a user I might want to ensure that `Kokkos` will use the `SYCL` specialization, and so maybe the user would like to provide `KOKKOS_ENABLE_IMPL_SYCL_GRAPH=ON` to ensure that `Kokkos` errors out if this cannot be done?

Reply:
`SYCL_EXT_ONEAPI_GRAPH` is defined within `SYCL` headers, so I need `KOKKOS_IMPL_HAVE_SYCL_EXT_ONEAPI_GRAPH` to discard the test. I think I prefer discarding the `TestSYCL_InterOp_Graph.cpp`, rather than haing it but empty.

Reply:
Closing.